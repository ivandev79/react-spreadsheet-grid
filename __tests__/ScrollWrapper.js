import React from 'react';
import renderer from 'react-test-renderer';
import ScrollWrapper from './../src/scrollWrapper';
import { columnsWithTextValues,
    columnsWithComponents,
    rows } from './../testKit'

test('ScrollWrapper without rows', () => {
    const tree = renderer.create(
        <ScrollWrapper
            columns={columnsWithTextValues}
            getRowKey={row => row.id}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('ScrollWrapper with columns with text values', () => {
    const tree = renderer.create(
        <ScrollWrapper
            columns={columnsWithTextValues}
            rows={rows}
            getRowKey={row => row.id}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

test('ScrollWrapper with columns with components', () => {
    const tree = renderer.create(
        <ScrollWrapper
            columns={columnsWithComponents}
            rows={rows}
            getRowKey={row => row.id}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});

