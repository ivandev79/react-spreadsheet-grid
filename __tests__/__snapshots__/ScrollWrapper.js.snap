// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollWrapper with columns with components 1`] = `
<div
  className="SpreadsheetTableScrollWrapper"
  onScroll={[Function]}
>
  <div
    className="SpreadsheetTableScrollDummy"
    style={
      Object {
        "height": "184px",
      }
    }
  />
  <table
    className="SpreadsheetTable"
    style={
      Object {
        "top": "0px",
      }
    }
  >
    <thead
      style={
        Object {
          "display": "",
          "height": "40px",
        }
      }
    >
      <tr>
        <th>
          Name
        </th>
        <th>
          Second name
        </th>
        <th>
          Position
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="SpreadsheetTable__placeholder"
      >
        <td
          colSpan={3}
        >
          <div>
            There are no rows
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ScrollWrapper with columns with text values 1`] = `
<div
  className="SpreadsheetTableScrollWrapper"
  onScroll={[Function]}
>
  <div
    className="SpreadsheetTableScrollDummy"
    style={
      Object {
        "height": "184px",
      }
    }
  />
  <table
    className="SpreadsheetTable"
    style={
      Object {
        "top": "0px",
      }
    }
  >
    <thead
      style={
        Object {
          "display": "",
          "height": "40px",
        }
      }
    >
      <tr>
        <th>
          Name
        </th>
        <th>
          Second name
        </th>
        <th>
          Position
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="SpreadsheetTable__placeholder"
      >
        <td
          colSpan={3}
        >
          <div>
            There are no rows
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`ScrollWrapper without rows 1`] = `
<div
  className="SpreadsheetTableScrollWrapper"
  onScroll={[Function]}
>
  <div
    className="SpreadsheetTableScrollDummy"
    style={
      Object {
        "height": "40px",
      }
    }
  />
  <table
    className="SpreadsheetTable"
    style={
      Object {
        "top": "0px",
      }
    }
  >
    <thead
      style={
        Object {
          "display": "",
          "height": "40px",
        }
      }
    >
      <tr>
        <th>
          Name
        </th>
        <th>
          Second name
        </th>
        <th>
          Position
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="SpreadsheetTable__placeholder"
      >
        <td
          colSpan={3}
        >
          <div>
            There are no rows
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
